org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/EclipseProjectImporter.java

```diff
@@ -43,6 +45,13 @@ public boolean applies(IProgressMonitor monitor) throws CoreException {
 		if (directories == null) {
 			BasicFileDetector eclipseDetector = new BasicFileDetector(rootFolder.toPath(), DESCRIPTION_FILE_NAME)
 					.addExclusions("**/bin");//default Eclipse build dir
+			Preferences preferences = getPreferences();
+			for (IProject project : ProjectUtils.getAllProjects()) {
```

Comment:
we might need a `getAllProjects(boolean includeInvisibleProjects)` method

Reply:
Fixed.